action.yml

```diff
@@ -40,23 +48,23 @@ outputs:
     description: Used version number
   prerelease:
     description: true is the version points to pre-release, false otherwise
+  devrelease:
+    description: true is the version points to dev-release, false otherwise
 branding:
   color: yellow
   icon: upload
 runs:
-  using: composite
-  steps:
-  - name: Run
-    run: ${{ github.action_path }}/get_releasenote.py
-    shell: bash
-    env:
-      GITHUB_WORKSPACE: ${{ github.workspace }}
-      INPUT_CHANGES_FILE:  ${{ inputs.changes_file }}
-      INPUT_OUTPUT_FILE:  ${{ inputs.output_file }}
-      INPUT_NAME:  ${{ inputs.name }}
-      INPUT_VERSION:  ${{ inputs.version }}
-      INPUT_VERSION_FILE:  ${{ inputs.version_file }}
-      INPUT_START_LINE:  ${{ inputs.start_line }}
-      INPUT_HEAD_LINE:  ${{ inputs.head_line }}
-      INPUT_FIX_ISSUE_REGEX:  ${{ inputs.fix_issue_regex }}
-      INPUT_FIX_ISSUE_REPL:  ${{ inputs.fix_issue_repl }}
+  using: docker
+  image: Dockerfile
```

Comment:
Note that with this approach, every job using the action will build the image. I've seen some actions optimizing this by publishing the built images to registries and pointing this there instead. Notably this one: https://github.com/fedora-python/tox-github-action/blob/9f7d1ff/action.yaml#L14.

Reply:
Interesting.
Does Github cache the built docker image?
An explicit image registry complicates the release process. Not drastically but still extra steps.
The project was super-lightweight initially. Now it grows :(

Reply:
> Does Github cache the built docker image?

It does not. This is why I emphasized that this happens in every job.

> An explicit image registry complicates the release process. [...] The project was super-lightweight initially. Now it grows :(

Yep, this is why I mentioned it. I still haven't built good automation for this but I've been wanting to use this approach for some of my actions like PyPI upload for a while now.